File: test/org/jetbrains/android/dsl/tests/functional/SimpleListenerTest.java
package org.jetbrains.android.dsl.tests.functional;

import org.jetbrains.android.dsl.BaseGeneratorProps;
import org.jetbrains.android.dsl.Subsystem;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class SimpleListenerTest extends BaseFunctionalTest {

    private final String testDataFile = "org/jetbrains/android/dsl/tests/functional/SimpleListenerTest.kt";

    @Override
    @BeforeMethod
    public void setUp() throws Exception {
        super.setUp();
    }

    protected void initSettings(BaseGeneratorProps settings) {
        settings.setGenerateSimpleListeners(true);
    }

    @Test
    public void testSimpleListeners() throws Exception {
        runFunctionalTest(testDataFile, Subsystem.LISTENERS);
    }
}

File: test/org/jetbrains/android/dsl/functional/SimpleListenerTest.kt
package org.jetbrains.android.dsl.functional
import org.jetbrains.android.dsl.*
import org.testng.annotations.Test
public class SimpleListenerTest : AbstractFunctionalTest() {
    private val testDataFile = "SimpleListenerTest.kt"
    override fun initSettings(settings: BaseGeneratorConfiguration) {
        settings.files.add(KoanFile.LISTENERS)
        settings.tunes.add(ConfigurationTune.SIMPLE_LISTENERS)
    }
    [Test] public fun testSimpleListenerTestFor21() {
        runFunctionalTest(testDataFile, KoanFile.LISTENERS, "21")
    }
    [Test] public fun testSimpleListenerTestFor21s() {
        runFunctionalTest(testDataFile, KoanFile.LISTENERS, "21s")
    }
}
